### ISSUE TITLE
[Bug]: File manager is not launched when passing a directory to xdg-open, it opens another application

### ISSUE THREAD

**Issue by eduardosm** [flatpak/flatpak#5235]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak/issues) for a bug that matches the one I want to file, without success.
- [X] If this is an issue with a particular app, I have tried filing it in the appropriate issue tracker for the app (e.g. under https://github.com/flathub/) and determined that it is an issue with Flatpak itself.
- [X] This issue is not a report of a security vulnerability (see [here](https://github.com/flatpak/flatpak/blob/main/SECURITY.md) if you need to report a security issue).

### Flatpak version

1.14.0

### What Linux distribution are you using?

Ubuntu

### Linux distribution version

Kubuntu 22.10

### What architecture are you using?

x86_64

### How to reproduce

1. Launch a shell inside an app's sandbox (`flatpak run --command=sh <someapp>`)
2. Run `xdg-open $XDG_DATA_HOME`

If an application has an "open containing directory" or similar function, it can also be used to trigger the bug (for example, Steam's "show game data directory").

### Expected Behavior

The default file manager (e.g., Dolphin in KDE) should open.

### Actual Behavior

Another application opens (Visual Studio Code in my case).

### Additional Information

Running `xdg-open` outside flatpak opens the file manager correctly.

I did not install Visual Studio Code with flatpak.

**Comment by Asocius** [comment]
I also experience this bug, everytime a flatpak-App wants to launch the file explorer, it launches Kate (the KDE notepad) instead. I'm on KDE Plasma 5.26 and Flatpak 1.12.7

**Comment by Mikenux** [comment]
I cannot reproduce it in Fedora Workstation 37 (GNOME 43.2, flatpak 1.14.1).

**Comment by AdrianVovk** [comment]
I don't think this is a bug with Flatpak. It just means that you have
VS Code/Kate/Whatever other app as the default MIME handler for type
`inode/directory`. Some distros don't set a default handler for this
mime type, which means that the "default" becomes the last app you
used to open that file type.

On GNOME, you can permanently fix this with `gio mime inode/directory
org.gnome.Nautilus.desktop`. This might work on KDE as well (if you
change the .desktop entry to Dolphin's), or there is some KDE
equivalent command. You can query available apps that can open
directories via `gio mime inode/directory` or the equivalent KDE
command

**Comment by eduardosm** [comment]
`gio` already lists Dolphin as default aplication:
```
> gio mime inode/directory
Default application for “inode/directory”: org.kde.dolphin.desktop
Registered applications:
        org.kde.dolphin.desktop
        code.desktop
        org.gnome.baobab.desktop
        org.kde.gwenview.desktop
        org.kde.kate.desktop
Recommended applications:
        org.kde.dolphin.desktop
        code.desktop
        org.gnome.baobab.desktop
        org.kde.gwenview.desktop
        org.kde.kate.desktop
```
Dolphin also appears as default file manager in KDE settings.

Running `xdg-open` outside Flatpak correctly launches Dolphin when it passed a directory. This issue only happens with Flatpak applications.

**Comment by McCloud1710** [comment]
Same issue here. Every time i need to open a folder using a flatpak application, like Bottlesm kate is open, instead the dolphin's file explorer.

**Comment by rizitis** [comment]
`cat /usr/share/applications/mimeinfo.cache | grep inode/directory`
what is the output here? if its not dolphin in the first place then thats the reason...
You can edit this file and set indode/directory=org.kde.dolphin.desktop but on reboot might be overwritten again by dbus... if so, then use a `sed -i s/what.ever.desktop/org.kde.dolphin.desktop/g  /usr/share/applications/mimeinfo.cache` command on your start up to revert org.kde.dolphin.desktop in first place....

**Comment by eduardosm** [comment]
I think the MIME cache just specifies which applications can handle each MIME type, which looks like they are sorted alphabetically. "Hotfixing" the cache each time it is regenerated is not the solution to this issue. Flatpak should respect default application settings, like non-Flatpak applications do.

**Comment by rizitis** [comment]
I agree 100% that "Hotfixing" the cache each time it is regenerated is not the solution to this issue. I only want to say that this is not a flatpak issue. 
I will try to explain it better, sorry for my bad English. 

If dbus doesn't find a service running when requested by some application (flatpak or non-flatpak), then it tries to auto-spawn it. If for example both thunar and dolphin are installed, one of the two corresponding services is then launched.
 A bugzilla, still open after [19 years ...](https://bugzilla.mozilla.org/show_bug.cgi?id=266600)
In summary: `multiple other filemanagers installed that have a .service file for the org.freedesktop.FileManager1 
dbus interface (e.g. having both XFCE's Thunar and KDE's Dolphin installed) may/will result 
in one of those filemanagers being executed. In my case dolphin was consistently used`

Explaining  my case (the reason I m here...), in  a fresh installation with KDE Plasma, XFCE and Gnome-45 I had issues with flatpak and non-flatpak apps when xdg-open needed to open default file manager (Nautilus for me). I solved all problems by creating/edit  a  ~/.config/mimeapps.list for every user in my system `inode/directory=org.gnome.Nautilus.desktop` BUT still some times after reboot the problem was back again  except  flatpak installed app "Git Hub Desktop" which never respect my default file manager no matter reboot or not. In place of Nautilus it opened Dolphin! 
Then I realized that flatpak apps was reading  `/usr/share/applications/mimeinfo.cache` were dolphin was in first place. When I removed dolphin from there it didnt respect Nautilus which now was in first place, but finally did it after  `gio mime inode/directory org.gnome.Nautilus.desktop` . So in my case the solution was to complete remove the MIME cache from my system and everything works fine only reading my  ~/.config/mimeapps.list (flatpak or non flatpak apps). What i m trying to say is that this bug is not flatpak bug.

**Comment by eduardosm** [comment]
In my case, Dolphin is the only application that offers the `org.freedesktop.FileManager1` interface and, unlike your case, I am having this issue only with Flatpak applications.

**Comment by chrisawi** [comment]
Please run `/usr/libexec/xdg-desktop-portal -rv` in a terminal and then try opening a directory. What output do you see?
